src/kernel.cpp

```diff
@@ -373,7 +373,17 @@ bool ComputeNextStakeModifier(const CBlockIndex* pindexPrev, uint64_t& nStakeMod
 // Get stake modifier checksum
 unsigned int GetStakeModifierChecksum(const CBlockIndex* pindex)
 {
-    assert (pindex->pprev || pindex->GetBlockHash() == (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet));
+    if (pindex->pprev == nullptr) 
+    {
+        if(pindex->GetBlockHash() != (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet)) 
+        {
+            //Error on non-genesis blocks that don't have a previous block
+            throw std::runtime_error(
+                "Error: blockchain data corrupted.\n" 
+                "Go to the wallet's data directory and delete the folder txleveldb and the files starting with blk\n" 
```

Comment:
Let's be a little more specific here. I think most people would know what blk000*.dat means...

Reply:
You also might want to inform people that this will result in syncing from zero and that their balance will temporarily show zero.

Reply:
I think using x be more well understood. Made a commit that says `blk000x.dat (x is any numer)`. Added the part about syncing